packages/libs/components/src/plots/Node.tsx

```diff
@@ -60,28 +67,48 @@ export function NodeWithLabel(props: NodeWithLabelProps) {
     textAnchor = 'end';
   }
 
+  // mouse hover state
+  const [hover, setHover] = useState(false);
+
   return (
-    <g onClick={onClick}>
-      <DefaultNode
-        r={nodeRadius}
-        fill={color ?? DEFAULT_NODE_COLOR}
-        stroke={stroke ?? DEFAULT_STROKE}
-        strokeWidth={strokeWidth ?? DEFAULT_STROKE_WIDTH}
-        className="NodeWithLabel_Node"
-      />
-      {/* Note that Text becomes a tspan */}
-      <Text
-        x={textXOffset}
-        textAnchor={textAnchor}
-        fontSize={fontSize}
-        verticalAnchor="middle"
-        fontWeight={fontWeight}
-        fill={labelColor}
-        className="NodeWithLabel_Label"
-      >
-        {label && truncateWithEllipsis(label, truncationLength)}
-      </Text>
-      <title>{label}</title>
-    </g>
+    <>
+      <g onClick={onClick}>
+        <DefaultNode
+          r={nodeRadius}
+          fill={color ?? DEFAULT_NODE_COLOR}
+          stroke={stroke ?? DEFAULT_STROKE}
+          strokeWidth={strokeWidth ?? DEFAULT_STROKE_WIDTH}
+          className="NodeWithLabel_Node"
+          // hover event
+          onMouseEnter={() => setHover(true)}
+          onMouseLeave={() => setHover(false)}
+        />
+        {/* Note that Text becomes a tspan */}
+        <Text
+          x={textXOffset}
+          textAnchor={textAnchor}
+          fontSize={fontSize}
+          verticalAnchor="middle"
+          fontWeight={fontWeight}
+          fill={labelColor}
+          id="NodeLabelText"
+          style={{
+            cursor: 'ponter',
+            zIndex: 1000,
+            backgroundColor: 'red',
+            display:
+              label && showNodeLabels && selectedNodeLabels.includes(label)
```

Comment:
i think we can move this logic outside of the component and then simplify the props. Each `NodeWithLabel` operates on one node, so we could just do one prop called `showLabel` which is true if `showNodeLabels && selectedNodeLabels.includes(label)` is true and false otherwise. What do you think about that?

Reply:
@asizemore as suggested, it is now passed as a prop from the parent component, `NetworkPlot`